### PR TITLE
websocket: Adding IgnoreOrigin and minor bug fix

### PR DESCRIPTION
**Pull request by glaslos**
Run into an issue with the websocket server Origin default check. So added a flag to disable it. Also fixed an issue in case a connection upgrade fails.

### CODE REVIEW

**Comment by arnecls** [comment]
@andygrunwald have a second look please.
Origin flag reference: https://godoc.org/github.com/gorilla/websocket#hdr-Origin_Considerations
I'm fine with handling it this way.

File: docs/producers/websocket.rst [hunk 1]
```diff
@@ -97,3 +101,4 @@ Example
 	    Address: ":81"
 	    Path:    "/"
 	    ReadTimeoutSec: 3
+        IgnoreOrigin: false
```
Comment (andygrunwald) [inline docs/producers/websocket.rst]: It looks that the indenting does not fit. Don't know (yet) if this will end in a different rendering result.

**Comment by andygrunwald** [comment]
@arnecls Go for it.

**Comment by glaslos** [comment]
Yes, my bad, messed up the formatting in the Readme example

**Pull request merged**